File: StudentTraking.AngularWeb/TrakingStudent/App/Services/StaffService.js
(function (module) {
    'use strict';

    var StaffService = function ($http, $rootScope, Constant) {
        var service = {};
        var _constant = new Constant();
        service.StaffDetails = {
            Id: null,
            StaffId: "",
            StaffName: "",
            DepartmentId: null,
            PrimaryTagId: null,
            ReportingEmailId: "",
            StaffMobileNo: "",
            SchoolId: null,
            ClassId: null,
            SectionId: null,
            Address1: "",
            Address2: "",
            Country: "",
            State: "",
            City: "",
            ZipCode: "",
            Gender: "",
            DateOfBirthh: null,
            PrimaryTagDetail: "",
            DepartmentName: "",
            SchoolName: "",
            ClassName: "",
            SectionName: "",
            CountryName: ""
        };

        var getDepartmentList = function (branchId) {
            var APIURL = _constant.get("studenttrakingurl");
            var url = APIURL + _constant.get("departmentlist") + $rootScope.User.SecurityToken;
            return $http({
                url: url,
                method: "GET"
            })
        };

        var getStaffList = function (branchId) {
            var APIURL = _constant.get("studenttrakingurl");
            var url = APIURL + _constant.get("stafflist") + $rootScope.User.SecurityToken + '/' + branchId;
            //var params = { countryId: countryid };
            return $http({
                url: url,
                method: "GET"
                //params: params
            })
        };

        var getStaffListBySearch = function (searchObj) {
            var APIURL = _constant.get("studenttrakingurl");
            var url = APIURL + _constant.get("stafflist") + $rootScope.User.SecurityToken + '/' + $rootScope.User.SchoolId;
            if (searchObj.name != "") {
                url = url + '/' + searchObj.name;
            }
            url = url + '/' + searchObj.departmentId;

            return $http({
                url: url,
                method: "GET"
            })
        };

        var addStaff = function (objStaff) {
            var StaffSaveRequest = {
                SecurityToken: $rootScope.User.SecurityToken,
                Staff: objStaff
            };
            var APIURL = _constant.get("studenttrakingurl");
            var url = APIURL + _constant.get("savestaff");
            var params = {};
            return $http({
                url: url,
                method: "POST",
                data: StaffSaveRequest
            })
        };

        var deleteStaff = function (staffid) {
            var APIURL = _constant.get("studenttrakingurl");
            var url = APIURL + _constant.get("deletestaff");
            var requiredparams = { staffId: staffid, securityToken: $rootScope.User.SecurityToken };
            return $http({
                url: url,
                method: "GET",
                params: requiredparams
            })
        };

        service.getStaffList = getStaffList;
        service.getStaffListBySearch = getStaffListBySearch;
        service.addStaff = addStaff;
        service.getDepartmentList = getDepartmentList;
        return service;
    };
    module.service("StaffService", ['$http', '$rootScope', 'Constant', StaffService]);
}(angular.module('StudentTracking.staff')));

File: StudentTracking.Application.Main/StaffService.cs
using StudentTracking.Application.API;
using StudentTracking.Application.Models;
using StudentTracking.Data;
using System;
using System.Collections.Generic;
using StudentTracking.Application.Main.Extensions;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Data.Entity;
using StudentTracking.Application.Models.Models;

namespace StudentTracking.Application.Main
{
    public class StaffService : IStaffService
    {
        private StudentTrackingContext _dbContext = null;
        public StaffService(StudentTrackingContext cntx)
        {
            this._dbContext = cntx;
        }

        public IEnumerable<StaffModel> GetAll(int schoolId)
        {
            var entities = this._dbContext.StaffDetails.Where(s => s.SchoolId == schoolId).ToList();

            if (null != entities && entities.Count > 0)
                return entities.MapAsCollection<StaffDetail, StaffModel>();

            return new List<StaffModel>();
        }

        public IEnumerable<DepartmentModel> GetDepartment()
        {
            var entities = this._dbContext.Departments.ToList();

            if (null != entities && entities.Count > 0)
                return entities.MapAsCollection<Department, DepartmentModel>();

            return new List<DepartmentModel>();
        }

        public IEnumerable<StaffModel> GetByDepartment(int schoolId, int deptId)
        {
            var entities = this._dbContext.StaffDetails.Where(s => s.SchoolId == schoolId && s.DepartmentId == deptId).ToList();

            if (null != entities && entities.Count > 0)
                return entities.MapAsCollection<StaffDetail, StaffModel>();

            return new List<StaffModel>();
        }
        public StaffModel Get(int staffId)
        {
            var entity = this._dbContext.StaffDetails.Where(s => s.ID == staffId).FirstOrDefault();

            if (null != entity)
                return entity.MapAs<StaffDetail, StaffModel>();

            return null;
        }

        public StaffModel Save(StaffModel model)
        {
            var isNew = true;

            var entity = this._dbContext.StaffDetails.Where(s => s.ID == model.ID).FirstOrDefault();

            if (null != entity)
                isNew = false;

            entity = _populateValues(entity, model);
            if (isNew)
                this._dbContext.StaffDetails.Add(entity);
            else
                this._dbContext.Entry<StaffDetail>(entity).State = EntityState.Modified;

            this._dbContext.SaveChanges();

            
            //Create staff account

            var securitySVC = new SecurityService(this._dbContext);
            var userModel = new UserModel
            {
                UserId = entity.StaffId,
                UserRole = (entity.DepartmentId == 5) ? "Teacher" : "Staff",
                Password = "Welcome1@",
                Name = entity.StaffName,
                SchoolId = entity.SchoolId,
                ClassId = entity.ClassId,
                SectionId = entity.SectionId,
                EmailId = entity.ReportingEmailId,
                ContactNumber = entity.StaffMobileNo
            };
            securitySVC.Save(userModel);

            return entity.MapAs<StaffDetail, StaffModel>();
        }

        private StaffDetail _populateValues(StaffDetail entity, StaffModel model)
        {
            if (null == entity)
            {
                entity = new StaffDetail();

                entity.ID = model.ID;
                entity.StaffId = model.StaffId;
                entity.SchoolId = model.SchoolId;
            }
            entity.ClassId = model.ClassId;
            entity.DepartmentId = model.DepartmentId;
            entity.PrimaryTagId = model.PrimaryTagId;
            entity.ReportingEmailId = model.ReportingEmailId;
            entity.SectionId = model.SectionId;
            entity.StaffMobileNo = model.StaffMobileNo;
            entity.StaffName = model.StaffName;
            entity.Address1 = model.Address1;
            entity.Address2 = model.Address2;
            entity.City = model.City;
            entity.Country = model.Country;
            entity.DateOfBirthh = model.DateOfBirthh;
            entity.DepartmentId = model.DepartmentId;
            entity.Gender = model.Gender;
            entity.StateId = model.StateId;
            entity.ZipCode = model.ZipCode;

            return entity;
        }
    }
}
